src/os/bluestore/BlueStore.cc

```diff
@@ -6138,8 +6134,6 @@ int BlueStore::_do_zero(TransContext *txc,
   // they may touch.
   o->flush();
 
-  o->bc.discard(offset, length);
```

Comment:
Yeah i have a trello card open for this.  We probably just need to make the wctx_finish code that drops lextent refs also invalidate the blob cache buffers (assuming they are clean).